### PR TITLE
Make intermediate build config available

### PR DESCRIPTION
**Pull request by loosebazooka**
The fewest changes to expose layers that I could think of.

### CODE REVIEW

File: jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerDescription.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.google.cloud.tools.jib.api;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+
+/**
+ * A class containing the representation of the contents of a container. Currently only exposes
+ * "layers", but can be extended to expose {@link ContainerConfiguration} or other informational
```
Comment (loosebazooka) [inline jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerDescription.java]: As this comment notes, ContainerConfiguration is left out... it's not in API, but maybe we can put it in API?

File: jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/skaffold/SyncMapMojoTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
+package com.google.cloud.tools.jib.maven.skaffold;
```
Comment (chanseokoh) [inline jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/skaffold/SyncMapMojoTest.java]: copyright

File: jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/SyncMapMojo.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
+package com.google.cloud.tools.jib.maven.skaffold;
```
Comment (chanseokoh) [inline jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/SyncMapMojo.java]: copyright

File: jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerDescription.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package com.google.cloud.tools.jib.api;
```
Comment (chanseokoh) [inline jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerDescription.java]: copyright

And add the new public API to jib-core CHANGELOG?

File: jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerDescription.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ */
+public class JibContainerDescription {
+
+  private final ImmutableList<LayerConfiguration> layers;
+
+  JibContainerDescription(List<LayerConfiguration> layers) {
+    this.layers = ImmutableList.copyOf(layers);
+  }
+
+  /** Returns a list of "user configured" layers, does *not* include base layer information. */
```
Comment (chanseokoh) [inline jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerDescription.java]: Ah, this is revealing. Maybe `getAddedLayers()` is more aptly named?

File: jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/skaffold/SyncMapMojoTest.java (hunk 2 above)
Comment (loosebazooka) [reply jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/skaffold/SyncMapMojoTest.java]: actually this file shouldn't be in here, git is acting all crazy on my computer

File: jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerDescription.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+  private final ImmutableList<LayerConfiguration> layers;
+
+  JibContainerDescription(List<LayerConfiguration> layers) {
+    this.layers = ImmutableList.copyOf(layers);
+  }
+
+  /**
+   * Returns a list of "user configured" layers, does *not* include base layer information.
+   * @return An {@link ImmutableList} of {@link LayerConfiguration}s
```
Comment (chanseokoh) [inline jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerDescription.java]: Doesn't `./gradlew javadoc` complain how having a newline before `@return`? And did you consider the name `getAddedLayers()`? I think it's lower the chance to mistakenly assume they include base image layers.

Comment (loosebazooka) [reply jib-core/src/main/java/com/google/cloud/tools/jib/api/JibContainerDescription.java]: yeah, I guess I hadn't rebased by then.

**Pull request merged**